File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/C++/s012761786_commented.cpp
#include <iostream>
#include <algorithm>
#include <vector>

int main() {
    long n, a, b;
    std::cin >> n >> a >> b;

    long sa = std::abs(a - b);

    long ans = 0;

    if (sa % 2 == 0) {
        ans = sa / 2;
    } else {
        long AAA = std::min((n - a), (a - 1));
        long BBB = std::min((n - b), (b - 1));

        ans = std::min(AAA, BBB); // One of them reaches the edge.

        // Decrease the difference by 1 => the difference becomes even
        sa -= 1;
        ans += 1;

        // The difference is now even
        ans += sa / 2;
    }

    std::cout << ans << std::endl;

    //std::cout << 1 / 2 << std::endl;

    return 0;
}

// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/inline/comment/direct/Java/Go/s012761786_commented.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var n, a, b int64
	fmt.Scan(&n, &a, &b)

	sa := int64(math.Abs(float64(a - b)))

	var ans int64

	if sa%2 == 0 {
		ans = sa / 2
	} else {
		AAA := int64(math.Min(float64(n-a), float64(a-1)))
		BBB := int64(math.Min(float64(n-b), float64(b-1)))

		ans = int64(math.Min(float64(AAA), float64(BBB))) // one side reaches the edge.

		// reduce difference by 1 => difference becomes even
		sa -= 1
		ans += 1

		// difference is now even
		ans += sa / 2
	}

	fmt.Println(ans)
}

// End of Code
